<a href="https://colab.research.google.com/github/Sj3604/Data-Science-Projects-/blob/main/Population.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
df = pd.read_csv("/content/state_wise_population__2019.csv")
print(df.head())
print(df.info())

   S.No           State  total_population  population_male  population_female
0     1     MAHARASHTRA          37722136         19961736           17760400
1     2   UTTAR PRADESH          25302925         13433369           11869556
2     3  ANDHRA PRADESH          18171615          9192368            8979247
3     4     WEST BENGAL          18063509          9357777            8705732
4     5         GUJARAT          17835049          9541688            8293361
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 33 entries, 0 to 32
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   S.No               33 non-null     int64 
 1   State              33 non-null     object
 2   total_population   33 non-null     int64 
 3   population_male    33 non-null     int64 
 4   population_female  33 non-null     int64 
dtypes: int64(4), object(1)
memory usage: 1.4+ KB
None


# Task
Analyze the population data by categorizing states into North, East, South, and West zones, calculate the total population for each zone, and list the states included in each zone.

**Reasoning**:
Define a dictionary to map states to zones based on geographical location.



In [2]:
state_to_zone = {
    'North': ['UTTAR PRADESH', 'DELHI', 'MADHYA PRADESH', 'RAJASTHAN', 'HARYANA', 'PUNJAB', 'J&K', 'UTTARAKHAND', 'CHANDIGARH', 'HIMACHAL PRADESH'],
    'East': ['WEST BENGAL', 'BIHAR', 'JHARKHAND', 'CHHATTISGARH', 'ORISSA', 'ASSAM', 'TRIPURA', 'MIZORAM', 'MANIPUR', 'MEGHALAYA', 'NAGALAND', 'ANDAMAN & NICOBAR ISLANDS'],
    'South': ['ANDHRA PRADESH', 'KARNATAKA', 'TAMIL NADU', 'KERALA', 'PONDY', 'TELANGANA'],
    'West': ['MAHARASHTRA', 'GUJARAT', 'GOA', 'DNH', 'SIKKIM']
}

**Reasoning**:
Define a function to map states to zones and apply it to the DataFrame to create a new 'Zone' column, then display the updated DataFrame.



In [3]:
def get_zone(state):
    for zone, states in state_to_zone.items():
        if state in states:
            return zone
    return 'Unknown Zone'
df['Zone'] = df['State'].apply(get_zone)
display(df.head())

S.No           State  total_population  population_male  population_female  \
0     1     MAHARASHTRA          37722136         19961736           17760400   
1     2   UTTAR PRADESH          25302925         13433369           11869556   
2     3  ANDHRA PRADESH          18171615          9192368            8979247   
3     4     WEST BENGAL          18063509          9357777            8705732   
4     5         GUJARAT          17835049          9541688            8293361   

    Zone  
0   West  
1  North  
2  South  
3   East  
4   West

**Reasoning**:
Print the total population for each zone and then iterate through the state_to_zone dictionary to list the states in each zone.



In [6]:
print("Total Population by Zone:")
display(zone_population)
print("\nStates in each Zone:")
for zone, states in state_to_zone.items():
    print(f"{zone}: {', '.join(states)}")

Total Population by Zone:


Zone
East     37912478
North    76301057
South    86152354
West     57639346
Name: total_population, dtype: int64


States in each Zone:
North: UTTAR PRADESH, DELHI, MADHYA PRADESH, RAJASTHAN, HARYANA, PUNJAB, J&K, UTTARAKHAND, CHANDIGARH, HIMACHAL PRADESH
East: WEST BENGAL, BIHAR, JHARKHAND, CHHATTISGARH, ORISSA, ASSAM, TRIPURA, MIZORAM, MANIPUR, MEGHALAYA, NAGALAND, ANDAMAN & NICOBAR ISLANDS
South: ANDHRA PRADESH, KARNATAKA, TAMIL NADU, KERALA, PONDY, TELANGANA
West: MAHARASHTRA, GUJARAT, GOA, DNH, SIKKIM


**Reasoning**:
Group the DataFrame by zone and sum the `population_male` and `population_female` columns to get the total population of men and women in each zone, then display the results.

In [7]:
zone_gender_population = df.groupby('Zone')[['population_male', 'population_female']].sum()
print("\nTotal Male and Female Population by Zone:")
display(zone_gender_population)


Total Male and Female Population by Zone:


population_male  population_female
Zone                                     
East          19717534           18194944
North         40447566           35853491
South         43476076           42676278
West          30573827           27065519

## Summary:

### Data Analysis Key Findings

*   The states were successfully categorized into four zones: North, East, South, and West.
*   The total population for the East zone is 37,912,478.
*   The total population for the North zone is 76,301,057.
*   The total population for the South zone is 86,152,354.
*   The total population for the West zone is 57,639,346.
*   The states included in each zone are as follows:
    *   North: UTTAR PRADESH, DELHI, MADHYA PRADESH, RAJASTHAN, HARYANA, PUNJAB, J&K, UTTARAKHAND, CHANDIGARH, HIMACHAL PRADESH
    *   East: WEST BENGAL, BIHAR, JHARKHAND, CHHATTISGARH, ORISSA, ASSAM, TRIPURA, MIZORAM, MANIPUR, MEGHALAYA, NAGALAND, ANDAMAN & NICOBAR ISLANDS
    *   South: ANDHRA PRADESH, KARNATAKA, TAMIL NADU, KERALA, PONDY, TELANGANA
    *   West: MAHARASHTRA, GUJARAT, GOA, DNH, SIKKIM

### Insights or Next Steps

*   The South zone has the highest total population among the four zones.
*   Further analysis could involve calculating the average population per state within each zone or analyzing other demographic factors by zone.


**Reasoning**:
Filter the DataFrame to include only Uttar Pradesh and Bihar, then calculate and display the ratio of male to female population for each state.

In [8]:
# Filter the DataFrame for Uttar Pradesh and Bihar
uttar_pradesh_bihar_df = df[df['State'].isin(['UTTAR PRADESH', 'BIHAR'])]

# Calculate the ratio of men to women for each state
uttar_pradesh_ratio = uttar_pradesh_bihar_df[uttar_pradesh_bihar_df['State'] == 'UTTAR PRADESH']['population_male'].iloc[0] / uttar_pradesh_bihar_df[uttar_pradesh_bihar_df['State'] == 'UTTAR PRADESH']['population_female'].iloc[0]
bihar_ratio = uttar_pradesh_bihar_df[uttar_pradesh_bihar_df['State'] == 'BIHAR']['population_male'].iloc[0] / uttar_pradesh_bihar_df[uttar_pradesh_bihar_df['State'] == 'BIHAR']['population_female'].iloc[0]

# Display the ratios
print(f"Ratio of men to women in Uttar Pradesh: {uttar_pradesh_ratio:.2f}")
print(f"Ratio of men to women in Bihar: {bihar_ratio:.2f}")

Ratio of men to women in Uttar Pradesh: 1.13
Ratio of men to women in Bihar: 1.13


## Overall Analysis Summary:

Based on the provided state-wise population data for 2019, we have performed the following analysis:

1.  **Zone-wise Population Analysis**:
    *   We categorized the states into four geographical zones: North, East, South, and West, based on a predefined mapping.
    *   We calculated the total population for each zone. The total populations were:
        *   East Zone: 37,912,478
        *   North Zone: 76,301,057
        *   South Zone: 86,152,354
        *   West Zone: 57,639,346
    *   We also listed the states included in each zone.
    *   Furthermore, we determined the total male and female populations separately for each zone:
        *   East Zone: Male - 19,717,534, Female - 18,194,944
        *   North Zone: Male - 40,447,566, Female - 35,853,491
        *   South Zone: Male - 43,476,076, Female - 42,676,278
        *   West Zone: Male - 30,573,827, Female - 27,065,519

2.  **Male to Female Ratio Analysis**:
    *   We calculated the ratio of men to women for all states in the dataset.
    *   Specifically, we compared the ratios for Uttar Pradesh (1.13) and Bihar (1.13), finding them to be quite similar.
    *   We also displayed the male-to-female ratio for every state in the dataset, allowing for a comprehensive comparison across all states.

**Key Insights**:

*   The South zone has the highest total population among the four zones analyzed.
*   The male-to-female ratios vary across states, with some states having a higher proportion of males to females and vice versa.
*   Further analysis could involve visualizing these findings, examining population density by zone or state, or exploring trends over time if historical data were available.

**Reasoning**:
Calculate the ratio of male to female population for all states in the DataFrame and display the results.

In [9]:
# Calculate the ratio of men to women for all states
df['male_to_female_ratio'] = df['population_male'] / df['population_female']

# Display the state and their respective ratios
display(df[['State', 'male_to_female_ratio']])

State  male_to_female_ratio
0                 MAHARASHTRA              1.123946
1               UTTAR PRADESH              1.131750
2              ANDHRA PRADESH              1.023735
3                 WEST BENGAL              1.074898
4                     GUJARAT              1.150521
5                   KARNATAKA              1.055390
6                  TAMIL NADU              1.005075
7                       DELHI              1.146584
8              MADHYA PRADESH              1.094869
9                   RAJASTHAN              1.106064
10                      BIHAR              1.127679
11                    HARYANA              1.148619
12                     PUNJAB              1.144019
13                  JHARKHAND              1.107060
14               CHHATTISGARH              1.056439
15                     ORISSA              1.081516
16                     KERALA              0.940337
17                        J&K              1.119288
18                      ASSAM              1.073187
19                UTTARAKHAND              1.116000
20                 CHANDIGARH              1.205861
21                      PONDY              0.967024
22                    TRIPURA              0.997721
23                    MIZORAM              0.971517
24                    MANIPUR              0.947639
25           HIMACHAL PRADESH              1.222138
26                  MEGHALAYA              0.959564
27                   NAGALAND              1.106484
28  ANDAMAN & NICOBAR ISLANDS              1.124279
29                     SIKKIM              1.123694
30                        DNH              1.690673
31                  TELANGANA              1.012626
32                        GOA              1.027432